base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/ScriptGeneratorJsonFileHandler.java

```diff
@@ -0,0 +1,234 @@
+				List<Map<String, String>> mappedParamToValue =  currentlyLoadedJSONFileContent.getParameterValues();
+				newMappedParameterToValues =  getMappedParamToValue(mappedParamToValue);
+			} else {
+				throw new ScriptDefinitionNotMatched("Current script definition does not match with the "
+						+ "script definition that was used to generate the selected data file");
+			}
+		} catch(JsonSyntaxException | JsonIOException | IOException e) {
+			LOG.error(e);
+		} 
+		
+		return newMappedParameterToValues;
+	}
+	
+	/**
+	 * Content of data file that is currently loaded by user.
+	 * @return content of currently loaded data file in string format.
+	 */
+	public String getCurrentlyLoadedDataFileContent() {
+		Gson gson = new GsonBuilder().setPrettyPrinting().create();
+		return gson.toJson(currentlyLoadedJSONFileContent);
+	}
+	
+	/**
+	 * Get Content from give filename. 
+	 * @param filePath path of file to get content from
+	 * @throws JsonSyntaxException JSON syntax incorrect
+	 * @throws JsonIOException trouble reading JSON file
+	 * @throws FileNotFoundException JSON file does not exist
+	 * @return object representation of JSON which is read from a file
+	 */
+	private ParametersConverter convertJSONtoObject(String filePath) throws JsonSyntaxException, JsonIOException, FileNotFoundException{
+		
+		// Check if version number match
+		try {
+			Gson gson = new Gson();
+			FileReader reader = new FileReader(filePath);
+			ParametersConverter jsonContent = gson.fromJson(reader, ParametersConverter.class);
+			reader.close();
+			if (jsonContent.getJSONFormatVersionNumber().equals(CURRENT_VERSION)) {
+				return jsonContent;
+			} else {
+				LOG.error("Json " + jsonContent.getJSONFormatVersionNumber() + " version not supported");
+				throw new UnsupportedOperationException(String.format("Data file version %s is not supported.\nSupported version %s"
+						, jsonContent.getJSONFormatVersionNumber(), CURRENT_VERSION)); 
+			}
+		} catch (NullPointerException e) {
+			LOG.error(e + ": Json version number is null");
+		} catch (IOException e) {
+			LOG.error(e);
+		}
+		
+		return null;
+	}
+		
+	/**
+	 * Translate from List<Map<String, String>> to List<Map<ActionParameter, String>>.
+	 * @param mappedParamToValue list of parameters mapped to its value
+	 * @return List of Map<ActionParameter, String> values
+	 */
+	private ArrayList<Map<ActionParameter, String>> getMappedParamToValue(List<Map<String, String>> mappedParamToValue) {
+		ArrayList<Map<ActionParameter, String>> newMappedParamToValue = new ArrayList<Map<ActionParameter, String>>();
+		for (Map<String, String> map : mappedParamToValue) {
+			Map<ActionParameter, String> tempMap = new HashMap<ActionParameter, String>();
+			for (Map.Entry<String, String> entry : map.entrySet()) {
+				tempMap.put(new ActionParameter(entry.getKey()), entry.getValue());
+			}
+			newMappedParamToValue.add(tempMap);
+		}
+		return newMappedParamToValue;
```

Comment:
There must be a more succinct way of doing this using streams.

Reply:
Now done using stream.